torch/csrc/cudnn/Conv.cpp

```diff
@@ -140,11 +179,48 @@ struct algorithm_search<cudnnConvolutionBwdDataAlgo_t> {
     return bwd_data_algos;
   }
 
-  static cudnnConvolutionBwdDataAlgoPerf_t findAlgorithm(cudnnHandle_t handle, const Convolution& conv) {
+  static cudnnConvolutionBwdDataAlgoPerf_t findAlgorithm(THCState* state,cudnnHandle_t handle, const Convolution& conv,
+							  void* in, void* out, void* wght) {
     int algoCount;
+    cudnnStatus_t err;
     cudnnConvolutionBwdDataAlgoPerf_t perfResults;
-    CHECK(cudnnFindConvolutionBackwardDataAlgorithm(handle, conv.wdesc.desc,
-        conv.odesc.desc, conv.cdesc.desc, conv.idesc.desc, 1, &algoCount, &perfResults));
+    cudnnConvolutionBwdDataAlgo_t algo[] = {
+         CUDNN_CONVOLUTION_BWD_DATA_ALGO_0,
+         CUDNN_CONVOLUTION_BWD_DATA_ALGO_1,
+         CUDNN_CONVOLUTION_BWD_DATA_ALGO_FFT,
+         CUDNN_CONVOLUTION_BWD_DATA_ALGO_FFT_TILING,
+         CUDNN_CONVOLUTION_BWD_DATA_ALGO_WINOGRAD,
+         CUDNN_CONVOLUTION_BWD_DATA_ALGO_WINOGRAD_NONFUSED
+    };
+    size_t max_ws_size = 0;
+    void *data = NULL; // workspace
```

Comment:
I'd use a `unique_ptr` with `[](void *data) { THCudaFree(state, data); }` as deleter. This would simplify the code a lot.